src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ToolStripActionListTests.cs

```diff
@@ -0,0 +1,177 @@
+        _mockSelectionService = new();
+        _mockComponentChangeService = new();
+        _toolStrip = new();
+        InitializeMocks();
+
+        _designer = new();
+        _designer.Initialize(_toolStrip);
+        _actionList = new(_designer);
+
+        void InitializeMocks()
+        {
+            Mock<ISite> mockSite = new();
+            mockSite.Setup(s => s.GetService(typeof(IDesignerHost))).Returns(_mockDesignerHost.Object);
+            mockSite.Setup(s => s.GetService(typeof(ISelectionService))).Returns(_mockSelectionService.Object);
+            mockSite.Setup(s => s.GetService(typeof(IComponentChangeService))).Returns(_mockComponentChangeService.Object);
+
+            _toolStrip.Site = mockSite.Object;
+
+            _mockDesignerHost.Setup(d => d.GetService(typeof(IComponentChangeService))).Returns(_mockComponentChangeService.Object);
+        }
+    }
+
+    public void Dispose()
+    {
+        _toolStrip.Dispose();
+        _designer.Dispose();
+    }
+
+    [Fact]
+    public void Constructor_InitializesFields()
+    {
+        _actionList.Should().NotBeNull();
+        _actionList.Should().BeOfType<ToolStripActionList>();
+        _actionList.Component.Should().Be(_toolStrip);
+    }
+
+    [Theory]
+    [BoolData]
+    public void AutoShow_GetSet_ReturnsExpected(bool value)
+    {
+        _actionList.AutoShow.Should().BeFalse();
+
+        _actionList.AutoShow = value;
+        _actionList.AutoShow.Should().Be(value);
+    }
+
+    [Theory]
+    [EnumData<DockStyle>]
+    public void Dock_GetSet_ReturnsExpected(DockStyle dockStyle)
+    {
+        _actionList.Dock.Should().Be(DockStyle.Top);
+
+        _actionList.Dock = dockStyle;
+        _actionList.Dock.Should().Be(dockStyle);
+    }
+
+    [Theory]
+    [EnumData<ToolStripRenderMode>]
+    public void RenderMode_GetSet_ReturnsExpected(ToolStripRenderMode renderMode)
+    {
+        if (renderMode == ToolStripRenderMode.Custom)
+        {
+            _actionList.Invoking(a => a.RenderMode = renderMode)
+             .Should().Throw<NotSupportedException>();
+        }
+        else
+        {
+            _actionList.RenderMode = renderMode;
+            _actionList.RenderMode.Should().Be(renderMode);
+        }
+    }
+
+    [Theory]
+    [EnumData<ToolStripGripStyle>]
+    public void GripStyle_GetSet_ReturnsExpected(ToolStripGripStyle gripStyle)
+    {
+        _actionList.GripStyle.Should().Be(ToolStripGripStyle.Visible);
+
+        _actionList.GripStyle = gripStyle;
+        _actionList.GripStyle.Should().Be(gripStyle);
+    }
+
+    [Fact]
+    public void GetSortedActionItems_ShouldIncludeExpectedItems_WhenConditionsAreMet()
+    {
+        DesignerActionItemCollection items = _actionList.GetSortedActionItems();
+        List<DesignerActionItem> itemList = items.Cast<DesignerActionItem>().ToList();
+
+        itemList.Should().ContainSingle(i => i is DesignerActionMethodItem && ((DesignerActionMethodItem)i).MemberName == "InvokeEmbedVerb");
```

Comment:
I believe this can be simplified
```suggestion
        itemList.Should().ContainSingle(i => i is DesignerActionMethodItem action && (action.MemberName == "InvokeEmbedVerb");
```
same can be applied in other places in this change